Sprint-1/stretch/aoc-2018-day1/solution.js

```diff
@@ -0,0 +1,39 @@
+// 〰️ Advent of Code - Day 1: Chronal Calibration
+// 〰️ This program calculates the resulting frequency by processing changes from a file (input.txt).
+// 〰️ Each line in the file represents a frequency change (e.g., +1, -2, +3).
+// 〰️ The goal is to sum all the changes to find the final frequency.
+
+
+// 〰️ Step 1: Import the 'fs' module to work with the file system.
+const fs = require('fs');
+
+// 〰️ Step 2: Read the content of the input.txt file.
+// 〰️ - We use the 'utf8' encoding to read the file as a string.
+// 〰️ - The callback function handles the file content or an error if something goes wrong.
+fs.readFile('./Sprint-1/stretch/aoc-2018-day1/input.txt', 'utf8', function (error, data) {
```

Comment:
Why specify the file path as `'./Sprint-1/stretch/aoc-2018-day1/input.txt'` instead of as `'input.txt'`?

Reply:
It depends in which folder you execute your script. It seems like you were executing your script in the "project" folder (the folder containing "Sprint-1").

`input.txt` would only work if one executes the script directly in the folder `Sprint-1/stretch/aoc-2018-day1/`.

If we can assume `input.txt` file is always in the same folder as the script, and want to be able to execute the script from any folder, we can construct the absolute path of `input.txt` in the script. ChatGPT can give a good explanation if you ask it "In JS, how to construct the absolute path of a file located in the folder where the executed script is?".

Reply:
Hi!
I have to admit, I really struggled to understand this topic at first. Without ChatGPT's explanation, I don’t think I would have ever figured it out! 😅 I’m still not entirely sure if I’ve understood everything correctly, but I’ve made the changes to the code based on what I believe was the right approach. Could you please check it now and confirm if it's correct?

Here’s what I’ve done:
- I added the path module, which helps construct the absolute file path in a way that works across different operating systems (Windows, macOS, Linux).
- I replaced the relative file path ./Sprint-1/stretch/aoc-2018-day1/input.txt with the absolute path using __dirname and path.join(), ensuring that the file is always found, no matter from which directory the script is executed.

I would really appreciate it if you could review the updated code and let me know if it's correct now!

Thank you! 😊

Reply:
`__dirname` is an environment variable that tells you the absolute path of the directory containing the currently executing file.

So the `filePath` of `input.txt` can be constructed as 
`const filePath = path.join(__dirname, 'input.txt');`

You can use `console.log()` to output the value of `filePath`.

Reply:
Hi!

Thank you so much for your explanation! I realize now that I made the file pathway too long and complicated. 🙈 
I simplified the code and added a console.log(filePath); to check if the file path is correct, and it seems to work perfectly!
This makes so much more sense now! 😊